### ISSUE TITLE
Insolated datasources per module

### ISSUE THREAD

**Issue by victorherraiz** [spring-projects/spring-modulith#250]
is there any way to create a diferent datasource per module (e.g. diferent driver,  server, db...) and insolate that resources to avoid access from different modules.

I could be very nice feature to prevent data coupling and prepare the module to be use as an independant micro-service if is necesary.

**Comment by eynoc-coder** [comment]
I recommend creating a Config class within an internal package for each module and using Spring configuration to set up a dedicated datasource for each module. This [https://www.baeldung.com/spring-data-jpa-multiple-databases](url) provides an example of how to do this. Additionally, you can write an ArchUnit rule to prevent each datasource from scanning entities outside of their respective modules.

It still an Springboot application so all the features present in the previous versions still there. You just have to adapt it using the modulith rules.

I guess that respond your question.

**Comment by odrotbohm** [comment]
Yeah, I guess using an `AbstractRoutingDataSource` with a key per module registered via an interceptor, decorating entries into the module should do the trick. I wonder if there's any more direct support you imagine from Spring Modulith, because unless I overlook anything, it all boils down to your persistence setup.

**Comment by victorherraiz** [comment]
@odrotbohm, nice idea. I was not aware of `AbstractRoutingDataSource` solution. Is there any context holder that contains information of the current module to implement AbstractRoutingDataSource? With that, implementing an `AbstractRoutingDataSource` should be easy.

Maybe a future datasource per module configuration could be a great option.

**Comment by cassis163** [comment]
It would be neat to add an example where modules use the database per module pattern, since using a modular monolithic architecture does not directly imply the usage of a specific way to do persistence (e.g. shared database pattern). Also, this is not something that is currently mentioned in the documentation. The docs do not mention different ways to use persistence.

I should add that I merely skimmed through them, so I potentially failed to read about it.

**Comment by kriehzoo** [comment]
Same thought here and it would be great to have an example for this :-)
